### PR TITLE
Add support for fontawesome for github actions

### PR DESCRIPTION
**Pull request by alexthecarp**
Change github actions to set up .npmrc with fontawesome token.

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -2,14 +2,20 @@ name: Deploy Styleguide
 
 on:
   push:
-    branches:
-      - master
+#    branches:
```
Comment (roblevintennis) [inline .github/workflows/deploy.yml]: Can we prune these comments?

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -2,14 +2,20 @@ name: Deploy Styleguide
...
+#      - master
 
 jobs:
   build:
     name: Build and Deploy
     runs-on: ubuntu-latest
     steps:
+      - name: Setup fontawesome dependencies
+          run: |
```
Comment (roblevintennis) [inline .github/workflows/deploy.yml]: I'm not sure why it's upset: `You have an error in your yaml syntax on line 14`

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -24,6 +18,12 @@ jobs:
           path: node_modules
           # Don't share with other workflows because we are using `--production=false`
           key: dev-dependencies-${{ hashFiles('**/yarn.lock') }}
+      - name: Setup fontawesome dependencies
```
Comment (roblevintennis) [inline .github/workflows/deploy.yml]: Does moving it here allow the node modules to stay cached thus speeding things up I'm guessing?

File: .github/workflows/deploy.yml (hunk 2 above)
Comment (roblevintennis) [reply .github/workflows/deploy.yml]: `You have an error in your yaml syntax on line 22` I'm guessing it's a tabs / spaces nuanced issue

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (alexthecarp) [reply .github/workflows/deploy.yml]: no, they are here for a reason. normally we want the branch protection, but for any change of this yml file, if i leave that in then it wont' run until i merge. SO I have to comment them out, get it working and finally, as a last step, I uncomment these before checking in,

File: .github/workflows/deploy.yml (hunk 2 above)
Comment (alexthecarp) [reply .github/workflows/deploy.yml]: uyeah.. ugh

File: .github/workflows/deploy.yml (hunk 3 above)
Comment (alexthecarp) [reply .github/workflows/deploy.yml]: Yeah it feels like a better place. Though I'm not sure.

**Pull request merged**